File: java/src/test/java/com/dddeurope/niuz/fixtures/InMemoryAuthors.java
package com.dddeurope.niuz.fixtures;

import com.dddeurope.niuz.entities.Author;
import com.dddeurope.niuz.repositories.AuthorRepository;

import java.util.HashMap;
import java.util.Map;

public class InMemoryAuthors implements AuthorRepository {
    final Map<String, Author> authors = new HashMap<>();

    @Override
    public void save(Author author) {
        authors.put(author.getId(), author);
    }

    @Override
    public Author getByAuthorId(String authorId) {
        return authors.get(authorId);
    }
}

File: dotnet/niuz.application.test/fixtures/InMemoryAuthors.cs
using System.Collections.Generic;
using niuz.application.entities;
using niuz.application.repositories;

namespace niuz.application.fixtures
{
    public class InMemoryAuthors : IAuthorRepository
    {
        private readonly Dictionary<string, Author> authors = new Dictionary<string, Author>();
        
        public void Save(Author author)
        {
            authors[author.Id] = author;
        }

        public Author GetByAuthorId(string authorId)
        {
            return authors[authorId];
        }
    }
}